### PR TITLE
Add support for a “secondary” Rendezvous Server

### PR DESCRIPTION
**Pull request by jisqyv**
The primary rendezvous server is used for the initial rendezvous,
whether using WebRTC or not (Legacy Mode). This change provides the
location for a secondary rendezvous server to use for the WebRTC
negotiation. This is only done if the Companion indicates that it
supports this feature (sets “r2” to true in the JSON object it
posts). Any information provided by the Companion can be used by the
primary rendezvous server to determine which secondary server to use.

Change-Id: I8706c4f0fe66a0e902595689896003feff9cdff7

### CODE REVIEW

**Comment by jisqyv** [comment]
Note: This code is currently running on ai2-test

**Comment by ewpatton** [comment]
@jisqyv None of my comments are blockers, so if you don't think they need to be addressed at this time feel free to merge this change.

File: appinventor/appengine/src/com/google/appinventor/client/Ode.java [hunk 1]
```diff
@@ -2474,10 +2474,28 @@ public void setWarnBuild(boolean secondBuildserver, boolean value) {
...
+   * the default version from YaVersion, but from the blocks that are
+   * used to build the Companion.
+   *
+   * @param server the domain name of the Rendezvous servers
+   * @param inclQRcode true to indicate that the rendezvous server domain name should be included in the QR Code
+   *
+   */
+  private native void setRendezvousServer(String server, boolean inclQRcode) /*-{
     top.rendezvousServer = server;
+    top.includereninqrcode = inclQRcode;
```
[the remark this answers is not in the document]
Comment (jisqyv) [reply appinventor/appengine/src/com/google/appinventor/client/Ode.java]: I'll fix this.

File: appinventor/components/src/com/google/appinventor/components/runtime/util/WebRTCNativeMgr.java [hunk 2]
```diff
@@ -237,9 +273,12 @@ private void Poller() {
...
       }
 
-      Log.d(LOG_TAG, "Poller() Called");
+      if (DEBUG) {
+        Log.d(LOG_TAG, "Poller() Called");
+        Log.d(LOG_TAG, "Poller: rendezvousServer2 = " + rendezvousServer2);
+      }
       HttpClient client = new DefaultHttpClient();
-      HttpGet request = new HttpGet("http://" + rendezvousServer + "/rendezvous2/" + rCode + "-s");
+      HttpGet request = new HttpGet("http://" + rendezvousServer2 + "/rendezvous2/" + rCode + "-s");
```
[the remark this answers is not in the document]
Comment (jisqyv) [reply appinventor/components/src/com/google/appinventor/components/runtime/util/WebRTCNativeMgr.java]: I'm going to leave this alone for now. I will fix this, and I will also fix the companion blocks to also hot hardcode in "http://" at the same time.

**Pull request merged**